ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/UpdateMutabilityInSignatureFix.fs

```diff
@@ -0,0 +1,71 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+
+type UpdateMutabilityInSignatureFix(error: ValueNotContainedMutabilityAttributesDifferError) =
+    inherit FSharpQuickFixBase()
+
+    let tryFindBindingSignature (implementationDeclaredName: string) (declaredElement: IFSharpMember) =
+        let parentDeclarations = declaredElement.ContainingType.GetDeclarations()
+
+        // Find the parent signature counter part
+        let signatureNameModuleDeclaration =
+            parentDeclarations
+            |> Seq.tryPick (fun d ->
+                match d with
+                | :? INamedModuleDeclaration as signatureModule when
+                    signatureModule.GetSourceFile().IsFSharpSignatureFile -> Some signatureModule
+                | _ -> None)
+
+        signatureNameModuleDeclaration
+        |> Option.bind (fun signatureModule ->
+            signatureModule.Members
+            |> Seq.tryPick (fun sigMember ->
+                match sigMember with
+                | :? IBindingSignature as bindingSignature ->
+                    match bindingSignature.HeadPattern with
+                    | :? ITopReferencePat as bindingNamePat
+                        // This is a bit lucky to work.
+                        // There must be something more clever we can try
+                        when bindingNamePat.DeclaredName = implementationDeclaredName ->
+                        Some bindingSignature
+                    | _ -> None
+                | _ -> None)
+        )
+
+    let tryFindImplementationBindingInfo (pat: ITopReferencePat) =
+        if isNull pat then None else
+        let topLevelBinding = TopBindingNavigator.GetByHeadPattern(pat)
+        if isNull topLevelBinding then None else
+        let declaredElement = topLevelBinding.DeclaredElement :?> IFSharpMember
+        if isNull declaredElement then None else
+        let mfv = declaredElement.Mfv
+        if isNull mfv then None else
+        if not mfv.HasSignatureFile then None else
+        Some (topLevelBinding, declaredElement)
+
+    override x.Text = "Update mutability in signature"
```

Comment:
We probably want to have the pattern name included here, especially if we want to enable it for nested patterns now or later.

Reply:
I've added the name in the pattern. Let me know if you had something else in mind.